### PR TITLE
[gltf] Use flat shading if normals are not included.

### PR DESCRIPTION
**Pull request by donmccurdy**
[From spec](https://github.com/KhronosGroup/glTF/tree/master/specification/2.0#meshes):

> Implementation note: When normals are not specified, client implementations should calculate flat normals.

This helps with models created by obj2gltf, which may not always include normals.

### CODE REVIEW

File: examples/js/loaders/GLTF2Loader.js [hunk 1]
```diff
@@ -2196,6 +2196,12 @@ THREE.GLTF2Loader = ( function () {
 
 						}
 
+						if ( geometry.attributes.normal === undefined ) {
+
+							material.shading = THREE.FlatShading;
```
Comment (mrdoob) [inline examples/js/loaders/GLTF2Loader.js]: I think this should now be;

```js
material.flatShaded = true;
```

**Comment by donmccurdy** [comment]
Good catch, thanks. If it's OK I would like to support both cases for now, to keep the loader backwards-compatible with r86.

**Comment by mrdoob** [comment]
Oh heh. Yep, that sound good.

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**